nuitka/utils/CStrings.py

```diff
@@ -28,34 +28,31 @@ def _encodePythonStringToC(value):
     """
     assert type(value) is bytes, type(value)
 
-    result = ""
+    special_chars = {92: r"\134", 9: r"\011", 13: r"\015", 10: r"\012", 63: r"\077", 34: r"\042"}
+    result = ['"']
     octal = False
-
     for c in value:
-        if str is bytes:
-            cv = ord(c)
-        else:
-            cv = c
-
-        if c in b'\\\t\r\n"?':
-            result += r"\%03o" % cv
+        cv = c
 
+        if c in special_chars:
```

Comment:
The special_chars idea is half-way. A prepared dictionary for all 256 values, on the module level, would be easy to make Python2/Python3 difference, and then translate things. If you mixed "octal" into the key and value of that dict, it would be better.

I would criticize that the idea is good, but the execution could be better.